warehouse/query-core/src/main/java/datawave/query/util/keyword/KeywordExtractor.java

```diff
@@ -0,0 +1,121 @@
+
+    /** maximum length for extracted keywords */
+    private int maxNGrams = YakeKeywordExtractor.DEFAULT_MAX_NGRAMS;
+
+    /** maximum number of keywords to extract */
+    private int keywordCount = YakeKeywordExtractor.DEFAULT_KEYWORD_COUNT;
+
+    /** the maximum score threshold for keywords (lower score is better) */
+    private float maxScoreThreshold = YakeKeywordExtractor.DEFAULT_MAX_SCORE_THRESHOLD;
+
+    /** the maximum number of characters to process as input for keyword extraction */
+    private int maxContentLength = YakeKeywordExtractor.DEFAULT_MAX_CONTENT_LENGTH;
+
+    /** the language that should be used to process the input text */
+    private YakeLanguage language = BaseYakeLanguage.ENGLISH;
+
+    YakeKeywordExtractor yakeKeywordExtractor;
+
+    public KeywordExtractor(List<String> preferredViews, Map<String,byte[]> foundContent, Map<String,String> iteratorOptions) {
+        this.preferredViews = preferredViews;
+        this.foundContent = foundContent;
+        parseOptions(iteratorOptions);
+
+        // todo: handle multilingual input and pass the appropriate language to the extractor.
+        //@formatter:off
+        yakeKeywordExtractor = new YakeKeywordExtractor.Builder()
+                .withMaxNGrams(minNGrams)
+                .withMaxNGrams(maxNGrams)
+                .withKeywordCount(keywordCount)
+                .withMaxScoreThreshold(maxScoreThreshold)
+                .withMaxContentLength(maxContentLength)
+                .withLanguage(language)
+                .build();
+        //@formatter:on
+    }
+
+    public void parseOptions(Map<String,String> iteratorOptions) {
+
+        if (iteratorOptions.containsKey(MIN_NGRAMS)) {
+            minNGrams = Integer.parseInt(iteratorOptions.get(MIN_NGRAMS));
+        }
+
+        if (iteratorOptions.containsKey(MAX_NGRAMS)) {
+            maxNGrams = Integer.parseInt(iteratorOptions.get(MAX_NGRAMS));
+        }
+
+        if (iteratorOptions.containsKey(MAX_KEYWORDS)) {
+            keywordCount = Integer.parseInt(iteratorOptions.get(MAX_KEYWORDS));
+        }
+
+        if (iteratorOptions.containsKey(MAX_SCORE)) {
+            maxScoreThreshold = Float.parseFloat(iteratorOptions.get(MAX_SCORE));
+        }
+
+        if (iteratorOptions.containsKey(MAX_CONTENT_CHARS)) {
+            maxContentLength = Integer.parseInt(iteratorOptions.get(MAX_CONTENT_CHARS));
+        }
+    }
+
+    public KeywordResults extractKeywords() {
+        KeywordResults results = EMPTY_RESULTS;
+        for (String viewName : preferredViews) {
+            for (Map.Entry<String,byte[]> foundEntry : foundContent.entrySet()) {
+                String[] parts = foundEntry.getKey().split(":");
+                if (parts.length > 0 && viewName.equals(parts[0])) {
+                    final byte[] encodedContent = foundEntry.getValue();
+                    final byte[] decodedContent = ContentKeyValueFactory.decodeAndDecompressContent(encodedContent);
```

Comment:
I think this is safe to call even if previously decoded/decompressed

Reply:
Will resolve this in the next round of changes